wdl-grammar/src/experimental/lexer/v1.rs

```diff
@@ -111,14 +111,22 @@ pub enum HeredocCommandToken {
     #[token("~")]
     Tilde,
 
-    /// An ending angle bracket.
+    /// A single close angle bracket (not the end).
     ///
-    /// When three of these tokens are sequentially encountered,
-    /// [morph][super::Lexer::morph] the lexer to use [Token].
-    ///
-    /// Otherwise, consider the token to be part of the
-    /// literal text.
+    /// This token is part of the literal text.
     #[token(">")]
+    SingleCloseAngle,
+
+    /// A double close angle bracket (not the end).
+    ///
+    /// This token is part of the literal text.
+    #[token(">>")]
+    DoubleCloseAngle,
```

Comment:
What's the purpose of these tokens? Shouldn't `>` and `>>` be treated as any other literal text?

Reply:
Due to `logos` not being able to support backtracking as it only looks at an input exactly once, we can't have a pattern which effectively says "only match an unescaped `>>>`".  It's why the regular expression for literal text within the command is `[^\\~>]+`, which stops matching text at the first `>` that is encountered.

Therefore we then have to account for a single `>`, a double `>>`, and finally a truly closing `>>>`. The first two are simply treated as part of the text when we're parsing the interpolation.

Reply:
Ok that makes sense. Thanks for explaining!